### PR TITLE
chore(pages): fixes bad logic in container permissions check

### LINKED ISSUE #10275: pages_container_permission_check() causes an infinite loop

### PR DESCRIPTION
**Pull request by mrclay**
Fixes #10275

### CODE REVIEW

**Comment by mrclay** [comment]
I re-implemented this from the 2.2 logic.

File: mod/pages/start.php [hunk 1]
```diff
@@ -367,19 +367,39 @@ function pages_write_permission_check($hook, $entity_type, $returnvalue, $params
  * @return bool
  */
 function pages_container_permission_check($hook, $entity_type, $returnvalue, $params) {
-	$container = elgg_get_page_owner_entity();
-	if ($container && $container->canWriteToContainer(0, 'object', 'page')) {
+	$container = elgg_extract('container', $params);
+	$user = elgg_extract('user', $params);
+	$subtype = elgg_extract('subtype', $params);
+
+	// check type/subtype
```
Comment (mrclay) [inline mod/pages/start.php]: FYI this wasn't in 2.2 but seems safe to do.

**Review by hypeJunction — APPROVED** [review]

**Comment by mrclay** [comment]
FYI I changed only the commit summary.

**Comment by hypeJunction** [comment]
Merging as it makes it impossible to work in the dev environment

**Pull request merged**